presto-main/src/main/java/com/facebook/presto/execution/scheduler/BucketNodeMap.java

```diff
@@ -44,5 +46,10 @@ public BucketNodeMap(ToIntFunction<Split> splitToBucket)
         return getAssignedNode(splitToBucket.applyAsInt(split));
     }
 
+    public final Optional<Boolean> getAssignedNodeCache(Split split)
```

Comment:
Curious: why is this an `Optional<Boolean>`? I thought whether a split is cacheable should be a `boolean` instead? In that case we might as well just change the method name to `isSplitCacheable`. `getAssignedNodeCache` is a bit confusing because it seems that we are returning a cache.